stdlib/public/runtime/SwiftObject.mm

```diff
@@ -1347,12 +1347,38 @@ static bool isObjCForUnownedReference(void *value) {
   return type;
 }
 
+static bool classObjectConformsToProtocol(id object, Protocol *protocol) {
```

Comment:
This is basically copied from @mikeash's example in the discussion, but I had to do two `getClass` operations to make it work.  I'm not sure I understand that.

Reply:
That doesn't look right. That will obtain the metaclass of `object` and then follow that superclass chain. `class_conformsToProtocol` should always say "no" here, until it gets to the root class and the weird stuff that happens there. Here's the implementation of `conformsToProtocol:` (taken from https://github.com/showxu/objc4/blob/75f57479d4416815ac43e81848db76d101f86eab/objc4/runtime/NSObject.mm#L2308):

```
- (BOOL)conformsToProtocol:(Protocol *)protocol {
    if (!protocol) return NO;
    for (Class tcls = [self class]; tcls; tcls = tcls->getSuperclass()) {
        if (class_conformsToProtocol(tcls, protocol)) return YES;
    }
    return NO;
}
```

`[self class]` will be equivalent to `object_getClass(self)` in the absence of any weird overrides.

Reply:
If you're inspecting/debugging this stuff, it can be hard to tell apart classes and metaclasses, since they have the same name and the same `po` output in lldb. You can use `image lookup -va` and look for the symbol name (metaclasses will say `OBJC_METACLASS`) or call `class_isMetaClass(class)` to differentiate at runtime.